dtfs-central-api/src/repositories/tfm-users-repo/tfm-users.repo.ts

```diff
@@ -0,0 +1,23 @@
+import { Collection, WithoutId, ObjectId } from 'mongodb';
+import { MONGO_DB_COLLECTIONS, TfmUser } from '@ukef/dtfs2-common';
+import { mongoDbClient } from '../../drivers/db-client';
+
+export class TfmUsersRepo {
+  /**
+   * Gets the tfm users collection
+   * @returns The collection
+   */
+  private static async getCollection(): Promise<Collection<WithoutId<TfmUser>>> {
```

Comment:
Yes - this actually comes from the `getCollection` type that we define in the `MongoDbClient` class, but there is a reason that we include `WithoutId` there too - it is due to the rules around inserting in mongo.

When you have the type `Collection<TSchema>`, `TSchema` is the type that is required as the document you are inserting. If `TSchema` has the `_id` field defined, then you are required to supply `_id` despite the fact that we should defer to mongo to insert the `_id` to avoid clashes (event though they are unlikely). As all of our DB models are wrapped in the `WithId` type, we must apply the `WithoutId` type to strip that field when we get the collection so that we can insert documents without a defined `_id`. The same rule does not apply to `find` operations however as they automatically wrap the `TSchema` with `WithId` as a document will always be returned with an `_id` field. For further reading, I'd recommend looking into the `mongodb.d.ts` declaration file under the `Collection` type which includes the dicussed overloads for all the methods discussed